packages/node-cli/src/core/CommandActionExecutor.js

```diff
@@ -83,10 +85,19 @@ module.exports = class CommandActionExecutor {
 				projectConfiguration: projectConfiguration,
 			});
 
+			if (!(actionResult instanceof ActionResult)) {
+				throw 'INTERNAL ERROR: Command must return an ActionResult object.';
+			}
+
+			if (actionResult._status === ActionResultStatus.ERROR) {
+				throw actionResult._error;
+			}
+
 			this._commandOutputHandler.showSuccessResult(actionResult, command.formatOutputFunc);
 
-			if (actionResult && actionResult.operationResult) {
-				const operationResult = actionResult.operationResult;
+			const actionResultContext = actionResult._context;
+			if (actionResultContext.operationResult) {
```

Comment:
This assumes all commands return an `operationResult` which I think it's not true. OperationResult is an implementation detail of each command, but that detail should not be reflected in the ActionResult. 
Each command should read the operationResult and determine if based on the internals, the ActionResult should be success or error.

I would suggest to have a helper to transform an operationResult into an actionResult, and be used inside commands so any code outside the command generator does not need to know about SDK or OperationResults. To generate an actionResult from an operationResult, we can do something like:
`SDKOperationResultsUtils.createActionResultFrom(operationResult)`
We already have this utils class.

Reply:
I have started to develop that the commands return actionResult, right now only deploy and create project to advance code review... I'll be working now in the rest of commands.

Reply:
Team decided to just use for the moment builders of ActionResult, since the helper at the end would add a extra layer over ActionResult